packages/block-editor/src/components/rich-text/index.js

```diff
@@ -296,6 +296,10 @@ function RichTextWrapper(
 			const hasPastedBlocks = pastedBlocks.length > 0;
 			let lastPastedBlockIndex = -1;
 
+			// Consider the after value to be the original it is not empty and
+			// the before value *is* empty.
+			const isAfterOriginal = isEmpty( before ) && ! isEmpty( after );
```

Comment:
Hmmmm I'm not sure if I'm right but that means if we create a new block by pressing "return" we actually add a new block before the original one which is kinda confusing.

On the other hand, I'm wondering if it solves the issue from [here ](https://github.com/WordPress/gutenberg/issues/26252#issuecomment-716616579) 
 especially this point:
 
 - If there was anything attached to that entities' id, like metadata, that won't be there anymore. Or if anything references those ids, that tracking will be lost.
 
 
I think that the `clientID` should stay the same only for the true original block (the one that is selected when we hit the "return" key)

Reply:
I think that makes sense. Nothing changes about the second block in that case and you insert a block before the selected block essentially. Think about a heading. If you press enter before the heading, you expect the heading to stay intact and a paragraph before the heading.

Reply:
Just checked how it works in google docs and if you press enter at the beginning of the heading, the heading style is still there but I'm not a UX guy tho. However, I would love to hear more opinions about that.